server/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -57,6 +81,8 @@
      * Return a copy of the list of the values for the current document.
      */
     public final List<T> getValues() {
+        deprecated("ScriptDocValues#getValues", "Deprecated getValues used, values should be accessed directly "
+                + "(e.g., doc.foo.bar instead of doc.foo.values.bar)");
```

Comment:
This example doesn't make sense. The deprecation is because `doc['foo']` *is* a List, so no need to call `.values` to get a list of the values. There is no `bar` member. Also, please use the bracket notation for accessing the field from the doc map, as it will always work, while dot only works sometimes depending on the field name.

Reply:
Thanks! How about "the field is already a list and should be accessed directly (e.g., use doc['foo'] instead of doc['foo'].values)"?

Reply:
I like giving an example here because I think it is confusing without one. What about "Deprecated getValues used, the field is a list and should be accessed directly. For example, use doc['field_name'] instead of doc['field_name'].values."?